src/mavoscript.js

```diff
@@ -29,11 +29,19 @@ var _ = Mavo.Script = {
 			if (Array.isArray(a)) {
 				result = [];
 				var max = Math.max(a.length, b.length);
+				var leftIdentity = o.leftIdentity === undefined ? o.identity : o.leftIdentity;
+				var rightIdentity = o.rightIdentity === undefined ? o.identity : o.rightIdentity;
+
 				for (let i = 0; i < max; i++) {
-					result[i] = o.scalar(
-						a[i] === undefined ? o.identity : a[i],
-						b[i] === undefined ? o.identity : b[i]
-					);
+					if (a[i] === undefined) {
+						result[i] = o.leftScalar ? o.scalar(leftIdentity, b[i]) : o.leftScalar(b[i]);
```

Comment:
The check is reversed, you are applying `o.leftScalar` when it's not there. Did you test this code?!

Reply:
My bad, thanks for checking.